corehq/apps/api/resources/v0_5.py

```diff
@@ -278,100 +274,14 @@ def obj_delete(self, bundle, **kwargs):
-                                UserChangeMessage.phone_numbers_added(list(numbers_added))["phone_numbers"]
-                            )
-                            user_change_logger.add_change_message({'phone_numbers': change_messages})
-
-                elif key == 'groups':
-                    group_ids = bundle.data.get("groups", [])
-                    groups_updated = bundle.obj.set_groups(group_ids)
-                    if user_change_logger and groups_updated:
-                        groups = []
-                        if group_ids:
-                            groups = [Group.wrap(doc) for doc in get_docs(Group.get_db(), group_ids)]
-                        user_change_logger.add_info(UserChangeMessage.groups_info(groups))
-                elif key == 'email':
-                    lowercase_value = value.lower()
-                    if user_change_logger and getattr(bundle.obj, key) != lowercase_value:
-                        user_change_logger.add_changes({key: lowercase_value})
-                    setattr(bundle.obj, key, lowercase_value)
-                elif key == 'password':
-                    domain = Domain.get_by_name(bundle.obj.domain)
-                    if domain.strong_mobile_passwords:
-                        try:
-                            clean_password(bundle.data.get("password"))
-                        except ValidationError as e:
-                            errors.append(e.message)
-                    bundle.obj.set_password(bundle.data.get("password"))
-                    if user_change_logger:
-                        user_change_logger.add_change_message(UserChangeMessage.password_reset())
-                elif key == 'user_data':
-                    try:
-                        original_user_data = bundle.obj.metadata.copy()
-                        bundle.obj.update_metadata(value)
-                        # check changes post update to account for any changes in update_metadata
-                        if user_change_logger and original_user_data != bundle.obj.user_data:
-                            user_change_logger.add_changes({'user_data': bundle.obj.user_data})
-                    except ValueError as e:
-                        raise BadRequest(str(e))
-                elif key in ['first_name', 'last_name', 'language']:
-                    if user_change_logger and getattr(bundle.obj, key) != value:
-                        user_change_logger.add_changes({key: value})
-                    setattr(bundle.obj, key, value)
-                else:
-                    errors.append(_('Attempted to update unknown field {}.').format(key))
+            try:
+                update(bundle.obj, key, value, user_change_logger)
+            except InvalidFormatException as e:
+                errors.append(_('{} must be a {}.'.format(key, e.expected_type)))
+            except UnknownFieldException:
+                errors.append(_('Attempted to update unknown field {}.').format(key))
```

Comment:
nit: quote field name
```suggestion
                errors.append(_("Attempted to update unknown field '{}'").format(key))
```

Reply:
Resolved in 9e1cc67a5aebe04c9aa498258f1b548173174f1e